src/waveform/waveform.cpp

```diff
@@ -199,6 +219,17 @@ void Waveform::readByteArray(const QByteArray& data) {
         m_data[i].filtered.mid = use_mid ? static_cast<unsigned char>(mid.value(i)) : 0;
         m_data[i].filtered.high = use_high ? static_cast<unsigned char>(high.value(i)) : 0;
     }
+    m_stemCount = waveform.signal_stems_size();
+    for (int s = 0; s < m_stemCount; ++s) {
+        const io::Waveform::Signal& stem = waveform.signal_stems(s);
+        bool stem_valid = stem.units() == io::Waveform::RMS;
+        for (int i = 0; i < dataSize; ++i) {
+            m_data[i].stems[s] = stem_valid && i < stem.value_size()
+                    ? static_cast<unsigned char>(stem.value(i))
+                    : 0;
+        }
+    }
```

Comment:
I cannot use `fill` here as this will override the all struct and set other `stemIdx` to `0` while they technically may have valid data.
I've reworked your suggestion with a for loop